### PR TITLE
FLUME-3345 Add Kudu Flume Sinks

### PR DESCRIPTION
**Pull request by granthenke**
This patch adds the Flume sinks that were previously maintained
in the Apache Kudu project.

### CODE REVIEW

**Comment by granthenke** [comment]
@bessbd can you please review?

**Comment by mpercy** [comment]
Hey Grant, I'm not sure how much time I can commit to this but just to clarify, did you have to change anything when pulling this out of Kudu and into Flume? Or is this just a direct move?

**Comment by mpercy** [comment]
Also was the RM able to generate and release the binary jar with the latest Kudu release?

**Comment by granthenke** [comment]
> Hey Grant, I'm not sure how much time I can commit to this but just to clarify, did you have to change anything when pulling this out of Kudu and into Flume? Or is this just a direct move?

Yes, this was effectively a copy paste. I did need to remove decimal support due to an older Avro version in Flume (which I didn't want to take on changing in the initial change). Otherwise it's just package changes. 

> Also was the RM able to generate and release the binary jar with the latest Kudu release?

Yes they were.

**Comment by mpercy** [comment]
> I did need to remove decimal support due to an older Avro version in Flume (which I didn't want to take on changing in the initial change). Otherwise it's just package changes.

Oh, bummer, but I likely would have done the same thing for now.

File: flume-ng-sinks/flume-ng-kudu-sink/src/test/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (mpercy) [inline flume-ng-sinks/flume-ng-kudu-sink/src/test/resources/log4j2.properties]: nit: I think (at least for consistency) this should be a log4j2.xml file like https://github.com/apache/flume/blob/trunk/flume-ng-sinks/flume-hdfs-sink/src/test/resources/log4j2.xml

File: pom.xml [hunk 2]
```diff
@@ -80,10 +80,11 @@ limitations under the License.
     <joda-time.version>2.9.9</joda-time.version>
     <junit.version>4.10</junit.version>
     <kafka.version>2.0.1</kafka.version>
+    <kudu.version>1.10.0</kudu.version>
```
Comment (mpercy) [inline pom.xml]: nit: mind putting this after kite (and switch hive to alphabetical for good measure?)

**Review by mpercy — COMMENTED** [review]
Looks good to me, modulo a couple nits. The tests passed on Travis, even though for some reason the morphlines tests failed, so everything with this commit looks pretty kosher to me.

From a high level view, I think it's more appropriate for this plugin to live with Flume instead of Kudu at this stage, especially since there is now a Kudu binary jar that allows this to be testable while living in the Flume tree.

If anybody else disagrees, please speak up here.

**Comment by granthenke** [comment]
@mpercy Can we merge this now?

**Comment by mpercy** [comment]
@granthenke I want to make sure this doesn't break the build. Unfortunately at work I can't download the non-Maven-central deps... would you mind checking for me that this patch didn't break morphlines somehow? What I was planning on doing was making sure the same errors that appear in the Jenkins and Travis errors on this precommit run appear on trunk without this patch, but I'd have to do that at home and I haven't had the time the past few days.

**Comment by granthenke** [comment]
I rebased this patch to make sure I had all the latest changes, I also push the PR #306 with a dummy change to verify the current build state.

**Comment by granthenke** [comment]
@mpercy Based on the results of #306 trunk is also failing with the morphine dependency issue.

**Comment by mpercy** [comment]
Hmm, yeah. Alright, let's merge this and defer figuring out what is wrong with morphlines, since it looks unrelated.

**Comment by mpercy** [comment]
Thanks for taking care of this, Grant!

**Pull request merged**